libnetwork/osl/interface_linux.go

```diff
@@ -368,6 +364,9 @@ func setInterfaceIP(nlh *netlink.Handle, iface netlink.Link, i *Interface) error
 
 func setInterfaceIPv6(nlh *netlink.Handle, iface netlink.Link, i *Interface) error {
 	if i.AddressIPv6() == nil {
+		if err := setIPv6(i.ns.path, i.DstName(), false); err != nil {
+			log.G(context.TODO()).WithError(err).Debug("failed to disable IPv6")
+		}
```

Comment:
```suggestion
```
All you have to do to fix #33099 is to not disable IPv6 on interfaces for arbitrary reasons. The non-existence of a routable IPv6 address on an endpoint does not imply that IPv6 should be disabled entirely.

> Furthermore, if I just want a link local address (and don't want to allocate routable addresses), there's no way to configure the default bridge network to do so. Enabling ipv6 requires that I have a range of addresses to assign, but all I want is link-local addresses as we got by default before.

If the user wants the link to not have an IPv6 link-local address, they can disable it with sysctls.

Reply:
Yes - I was aiming to preserve the old behaviour, where IPv6 was disabled on all interfaces unless they were connected to IPv6 networks. But it was broken behaviour, so I'll remove it.

Then, as you say, IPv6 link-local addresses will appear on most people's IPv4 interfaces - which might surprise some, but the sysctl stops it.